### PR TITLE
Move processor implementation

### PR DESCRIPTION
**Pull request by febo**
This PR moves the processor implementation from `lib.rs` to individual instruction files under `processor`.

### CODE REVIEW

**Review by danenbm — CHANGES_REQUESTED** [review]

File: programs/bubblegum/program/src/asserts.rs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+                .ok_or(BubblegumError::CreatorShareTotalMustBe100)?;
+        }
+        if total != 100 {
+            return Err(BubblegumError::CreatorShareTotalMustBe100.into());
+        }
+    }
+    Ok(())
+}
+
+pub fn replace_leaf<'info>(
```
[the remark this answers is not in the document]
Comment (febo) [reply programs/bubblegum/program/src/asserts.rs:57]: Removed, it was already on the `utils` module.

File: programs/bubblegum/program/src/asserts.rs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            merkle_tree: merkle_tree.clone(),
+            noop: log_wrapper.clone(),
+        },
+        authority_pda_signer,
+    )
+    .with_remaining_accounts(remaining_accounts.to_vec());
+    spl_account_compression::cpi::replace_leaf(cpi_ctx, root_node, previous_leaf, new_leaf, index)
+}
+
+pub fn append_leaf<'info>(
```
[the remark this answers is not in the document]
Comment (febo) [reply programs/bubblegum/program/src/asserts.rs:85]: Removed, it was already on the `utils` module.

File: programs/bubblegum/program/src/asserts.rs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    spl_account_compression::cpi::append(cpi_ctx, leaf_node)
+}
+
+pub fn cmp_pubkeys(a: &Pubkey, b: &Pubkey) -> bool {
+    sol_memcmp(a.as_ref(), b.as_ref(), PUBKEY_BYTES) == 0
+}
+
+pub fn cmp_bytes(a: &[u8], b: &[u8], size: usize) -> bool {
+    sol_memcmp(a, b, size) == 0
+}
```
[the remark this answers is not in the document]
Comment (febo) [reply programs/bubblegum/program/src/asserts.rs:114]: Removed, it was already on the `utils` module.

File: programs/bubblegum/program/src/asserts.rs [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+pub fn assert_owned_by(account: &AccountInfo, owner: &Pubkey) -> Result<()> {
+    if !cmp_pubkeys(account.owner, owner) {
+        //todo add better errors
+        Err(ProgramError::IllegalOwner.into())
+    } else {
+        Ok(())
+    }
+}
+
+pub fn get_asset_id(tree_id: &Pubkey, nonce: u64) -> Pubkey {
```
[the remark this answers is not in the document]
Comment (febo) [reply programs/bubblegum/program/src/asserts.rs:161]: Removed, it was already on the `utils` module. For now, I think it is ok to leave it there. Once we have more pda things, we can create a separate module.

**Review by danenbm — DISMISSED** [review]
LGTM thank you for splitting this up!

**Review by danenbm — DISMISSED** [review]

**Review by danenbm — APPROVED** [review]

**Pull request merged**